pysquared/state_of_health.py

```diff
@@ -21,154 +20,64 @@ class StateOfHealth:
-        self.boot_count: Counter = boot_count
-        self.burned_flag: Flag = burned_flag
-        self.brownout_flag: Flag = brownout_flag
-        self.fsk_flag: Flag = fsk_flag
-
-        self.state: OrderedDict[str, Any] = OrderedDict(
-            [
-                # init all values in ordered dict to None
-                ("system_voltage", None),
-                ("system_current", None),
-                ("solar_voltage", None),
-                ("solar_current", None),
-                ("battery_voltage", None),
-                ("radio_temperature", None),
-                ("radio_modulation", None),
-                ("microcontroller_temperature", None),
-                ("internal_temperature", None),
-                ("error_count", None),
-                ("boot_count", None),
-                ("burned_flag", None),
-                ("brownout_flag", None),
-                ("fsk_flag", None),
-            ]
-        )
-
-    def update(self):
-        """
-        Update the state of health
-        """
-        try:
-            self.state["system_voltage"] = self.system_voltage()
-            self.state["system_current"] = self.current_draw()
-            self.state["solar_voltage"] = self.solar_voltage()
-            self.state["solar_current"] = self.charge_current()
-            self.state["battery_voltage"] = self.battery_voltage()
-            self.state["radio_temperature"] = self.radio_manager.get_temperature()
-            self.state["radio_modulation"] = self.radio_manager.get_modulation()
-            self.state["microcontroller_temperature"] = microcontroller.cpu.temperature
-            self.state["internal_temperature"] = self.imu_manager.get_temperature()
-            self.state["error_count"] = self.logger.get_error_count()
-            self.state["boot_count"] = self.boot_count.get()
-            self.state["burned_flag"] = self.burned_flag.get()
-            self.state["brownout_flag"] = self.brownout_flag.get()
-            self.state["fsk_flag"] = self.fsk_flag.get()
-
-        except Exception as e:
-            self.logger.error("Couldn't acquire data for state of health", err=e)
-
-        self.logger.info("State of Health", state=self.state)
+        self._log: Logger = logger
+        self._sensors: tuple[
+            PowerMonitorProto | RadioProto | IMUProto | Flag | Counter | Processor, ...
+        ] = args
 
-    def system_voltage(self) -> float | None:
+    def get(self) -> OrderedDict[str, object]:
         """
-        Get the system voltage
-
-        :return: The system voltage in volts
-        :rtype: float | None
-
+        Get the state of health
         """
-        if self.battery_power_monitor is not None:
-            voltage: float = 0
-            try:
-                for _ in range(50):
-                    voltage += (
-                        self.battery_power_monitor.get_bus_voltage()
-                        + self.battery_power_monitor.get_shunt_voltage()
```

Comment:
In _this_ implementation these voltages are not being added/averaged together. We should talk about it though.